templates/admin/jqadm/product/item-media.php

```diff
@@ -73,9 +73,21 @@ class="btn btn-card-header act-delete fa" tabindex="<?= $this->get( 'tabindex' )
 								<input class="item-url" type="hidden"
 									v-bind:name="`<?= $enc->js( $this->formparam( ['media', '_idx_', 'media.url'] ) ) ?>`.replace('_idx_', idx)"
 									v-model="item['media.url']">
-								<img v-if="item['media.preview']" class="item-preview"
+								<img v-if="item['media.preview'] && (item['media.mimetype'].startsWith('image') || item['media.mimetype'].startsWith('video'))" class="item-preview"
 									v-bind:src="item['media.preview']"
 									v-bind:alt="item['media.label']">
+								<div v-else-if="item['media.mimetype'] === 'audio/mpeg'" class="audio-player">
+									<div v-if="item['media.url']">
+										<audio controls 
+											v-bind:src="'/uploads/tx_aimeos/' + item['media.url']"
```

Comment:
The URL to preview files is added here: https://github.com/aimeos/ai-admin-jqadm/blob/master/src/Admin/JQAdm/Product/Media/Standard.php#L436

The easiest way would be to use the URL also for the preview property for audio files when the file is uploaded. Everything else requires always some special handling which is not preferable.

Reply:
But this is not possible, because at the moment of creating a new media file the file itself is not yet uploaded (and the audio player uses a blob from the local file). But once the media file exists and is created from DB, there is no local blob anywhere, it only can use the the uploaded file.

So it is essential to realise that there are 2 szenarios for displaying a player: 
- the first one is the one when a user creates a new media item (local blob)
- the second one is, when an existing entry can be displayed already (taking the info from DB)

This is a bit different with what you do with the preview files, which are newly created images from canvas-toBlob(). I am not aware of anything similar for video or audio. If you know a way to use only preview, please post the code.

But this situation really, really need the separate upload path, which is "/uploads/tx-aimeos/" in TYPO3! How can I get that separately?

Reply:
You can get the path in the template using `$this->content('')`

Reply:
That does not return the upload path, but again an array. I am in the Standard.php file in the "fromArray" method (which should be the method that creates the media item entry?) and get what is shown in the screenshot.

Please: All I need it the upload path, which is "/uploads/tx-aimeos/" in TYPO3. In my opinion this must be attached to the entry['media...] array (in media.js), so it can be used for concatenation, which is needed to solve the local/remote file problem. Of course, only, if you agree with having such a solution in your Aimeos software. If not, please let me know!

![Screenshot 2023-03-10 at 11 15 31](https://user-images.githubusercontent.com/213803/224290638-d997d076-7529-46e5-9b67-e8f7eb640a44.png)

Reply:
And again i have to tell you that this is wrong! I need ONLY the upload path! Your solution is again NOT the solution! Not sure why this is so hard to understand?